File: leetcode/lc-us/cpp/2614.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int diagonalPrime(vector<vector<int>> &nums) {
        auto is_prime = [](int x) {
            if (x < 2) return false;
            for (int i = 2; i * i <= x; i++) {
                if (x % i == 0) return false;
            }
            return true;
        };
        int n = (int) nums.size(), res = 0;
        for (int i = 0; i < n; i++) {
            if (is_prime(nums[i][i])) res = max(res, nums[i][i]);
            if (is_prime(nums[i][n - 1 - i])) res = max(res, nums[i][n - 1 - i]);
        }
        return res;
    }
};

File: leetcode/lc-us/rust/2614.rs
use std::cmp::max;

impl Solution {
    fn is_prime(x: i32) -> bool {
        if x < 2 {
            return false;
        }
        let mut i = 2;
        while i * i <= x {
            if x % i == 0 {
                return false;
            }
            i += 1;
        }
        true
    }
    pub fn diagonal_prime(nums: Vec<Vec<i32>>) -> i32 {
        let n = nums.len();
        let mut res = 0;
        for i in 0..n {
            if Solution::is_prime(nums[i][i]) {
                res = max(res, nums[i][i]);
            }
            if Solution::is_prime(nums[i][n - 1 - i]) {
                res = max(res, nums[i][n - 1 - i]);
            }
        }
        res
    }
}